pipeline/pipeline_info.go

```diff
@@ -0,0 +1,215 @@
+package pipeline
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+	"time"
+
+	"github.com/ozontech/file.d/logger"
+	"go.uber.org/atomic"
+)
+
+//go:embed template
+var PipelineTpl embed.FS
+
+type pluginsObservabilityInfo struct {
+	In            inObservabilityInfo       `json:"in"`
+	Out           outObservabilityInfo      `json:"out"`
+	ActionPlugins []actionObservabilityInfo `json:"actions"`
+	LogChanges    LogChangesDTO             `json:"log_changes"`
+}
+
+type inObservabilityInfo struct {
+	PluginName string         `json:"plugin_name"`
+	WatchFiles bool           `json:"watch_files"`
+	FdCount    int64          `json:"fd_count"`
+	FileCount  int64          `json:"file_count"`
+	FDs        map[string]int `json:"fds"`
+}
+
+type outObservabilityInfo struct {
+	PluginName      string           `json:"plugin_name"`
+	BatcherCounters []batcherCounter `json:"batcher_counters"`
+	BatcherMinWait  BatcherTimeDTO   `json:"batcher_min_wait"`
+	BatcherMaxWait  BatcherTimeDTO   `json:"batcher_max_wait"`
+}
+
+type batcherCounter struct {
+	Seconds          int64 `json:"seconds"`
+	BatchesCommitted int64 `json:"batches_committed"`
+}
+
+type actionObservabilityInfo struct {
+	PluginName string              `json:"plugin_name"`
+	MetricName string              `json:"metric_name"`
+	Tracked    bool                `json:"tracked"`
+	Statuses   []actionEventStatus `json:"statuses"`
+}
+
+type actionEventStatus struct {
+	Name  string `json:"name"`
+	Count int64  `json:"count"`
+	Color string `json:"color"`
+}
+
+func (p *Pipeline) GetDeltas() *deltas {
+	inputEvents := newDeltaWrapper()
+	inputSize := newDeltaWrapper()
+	outputEvents := newDeltaWrapper()
+	outputSize := newDeltaWrapper()
+	readOps := newDeltaWrapper()
+
+	return p.incMetrics(inputEvents, inputSize, outputEvents, outputSize, readOps)
+}
+
+func (p *Pipeline) boardInfo(
+	inputInfo *InputPluginInfo,
+	actionInfos []*ActionPluginStaticInfo,
+	outputInfo *OutputPluginInfo,
+) (pluginsObservabilityInfo, error) {
+	result := pluginsObservabilityInfo{}
+	changes := p.LogChanges(p.GetDeltas())
+	changes.Interval *= time.Second
+
+	result.LogChanges = changes
+	in := inObservabilityInfo{
+		PluginName: inputInfo.Type,
+	}
+	inInfo, err := p.input.GetObservabilityInfo()
+	if err != nil {
+		return pluginsObservabilityInfo{}, err
+	}
+	if inInfo.WatcherInfo.IsValid {
```

Comment:
pipeline shouldn't know anything about input plugin details

Reply:
I removed this code. Now this logic hidden in plugin and html templates